src/gui/widgets/listbox.cpp

```diff
@@ -627,17 +636,19 @@ void listbox::set_active_sorting_option(const order_pair& sort_by, const bool se
 
 const listbox::order_pair listbox::get_active_sorting_option()
 {
-	const auto iter = std::find_if(orders_.begin(), orders_.end(),
-		[](const std::pair<selectable_item*, generator_sort_array>& option) {
-			return option.first != nullptr && option.first->get_value() != SORT_NONE;
+	unsigned int column = 0;
+
+	for(const auto& order : orders_) {
```

Comment:
Given that you need the index, it would be better to use a traditional for loop:

    for(unsigned int column = 0; column < orders_.size(); ++column) {